integration/python/test.py

```diff
@@ -57,21 +60,30 @@ def test_minio_lib():
         http_client=pool_manager,
     )
 
-    client.make_bucket("test-minio-python-lib")
-
-    with open("../testdata/small.txt", "rb") as f:
-        client.put_object("test-minio-python-lib", "small", f, 1)
-    with open("../testdata/large.txt", "rb") as f:
-        client.put_object("test-minio-python-lib", "large", f, 65*1024*1024)
-
-    res = client.list_objects_v2("test-minio-python-lib")
-    assert set((o.object_name, o.size) for o in res) == EXPECTED_KEYS
-
-    with open("../testdata/small.txt", "rb") as f:
-        assert client.get_object("test-minio-python-lib", "small").read() == f.read()
-    with open("../testdata/large.txt", "rb") as f:
-        assert client.get_object("test-minio-python-lib", "large").read() == f.read()
-
-    client.remove_object("test-minio-python-lib", "small")
-    client.remove_object("test-minio-python-lib", "large")
-    client.remove_bucket("test-minio-python-lib")
+    bucket_name = "test-minio-python-lib"
+    client.make_bucket(bucket_name)
+
+    try:
+        client.put_object(bucket_name, "small", SMALL_FILE.open("rb"), SMALL_FILE.stat().st_size)
+        client.put_object(bucket_name, "large", LARGE_FILE.open("rb"), LARGE_FILE.stat().st_size)
+
+        res = client.list_objects(bucket_name)
+        assert set((o.object_name, o.size) for o in res) == EXPECTED_KEYS
+
+        for (object_name, file) in (("small", SMALL_FILE), ("large", LARGE_FILE)):
+            object = client.stat_object(bucket_name, object_name)
+            response = client.get_object(bucket_name, object_name)
+            assert response.data == file.read_bytes()
+            try:
+                client.get_object(bucket_name, object_name, request_headers={"If-None-Match": object.etag})
```

Comment:
Nit, I know, but would it make sense to break this out into a separate test? I think it'd be fine to use the minio client for both (and even to re-use a lot of code for both) but I think it could be good to make it clear to future maintainers that we need unquoted etag support for reasons other than supporting the minio client (in fact, I'd comment the test saying that we need it for Spark support).

I'm just envisioning some senior eng new hire a year from now being like "Hah, these Pachydermies don't know anything about Minio, it doesn't do this" and pulling it out :joy:

Reply:
fair point! I wasn't trying to change up the structure too much but your reasoning is more than enough justification to do so